kumir/KumirLexer.g4

```diff
@@ -0,0 +1,154 @@
+// KumirLexer.g4
+// ANTLR v4 Lexer Grammar for the Kumir language.
+// Source: Refined based on official documentation and extensive testing
+//         against K.Y. Polyakov's examples. Developed collaboratively.
+// Author: [Your Name/GitHub Handle]
+// License: MIT License
+
+lexer grammar KumirLexer;
+
+// --- Keywords (Core Language) ---
+// Keywords are case-insensitive (both lowercase and uppercase Cyrillic are matched).
+MODULE              : ('модуль' | 'МОДУЛЬ');
```

Comment:
Are keywords fully case insensitive? I mean if the following is allowed

```g4
MODULE : 'МоДуЛь';
```

I suggest rewriting the keywords using [`caseInsensitive`](https://github.com/antlr/antlr4/blob/dev/doc/options.md#caseinsensitive) option:

```g4
options { caseInsensitive = true; }
...
MODULE : 'МОДУЛЬ';
```

Also, parentheses are redundant.